internal/tasks/account_management.go

```diff
@@ -97,16 +95,25 @@ func (j *Janitor) enforceManagedAccount(ctx context.Context, accountName models.
 	// the error returned from either tryDeleteManagedAccount or createOrUpdateManagedAccount is the main error that this method returns...
 	var nextCheckDuration time.Duration
 	if account == nil {
-		var done bool
-		done, err = j.tryDeleteManagedAccount(ctx, accountName)
-		switch {
-		case done:
-			nextCheckDuration = 0 // account has been deleted -> next check not necessary
-		case err == nil:
-			nextCheckDuration = 1 * time.Minute // we are making progress to delete this account -> recheck soon
-		default:
-			err = fmt.Errorf("could not delete managed account %q: %w", accountName, err)
-			nextCheckDuration = 5 * time.Minute // default interval for recheck after error
+		var accountModel *models.Account
+		accountModel, err = keppel.FindAccount(j.db, accountName)
+		if err != nil {
+			return err
+		}
+		if errors.Is(err, sql.ErrNoRows) {
+			nextCheckDuration = 0 // assume the account got already deleted
+		} else {
+			actx := keppel.AuditContext{
+				UserIdentity: janitorUserIdentity{TaskName: "account-sync"},
+				Request:      janitorDummyRequest,
+			}
+			err = j.processor().MarkAccountForDeletion(*accountModel, actx)
+			if err == nil {
+				nextCheckDuration = 0 // account will be deleted -> next check not necessary
```

Comment:
0 will not do what we want because that leaves the `next_enforcement_at` timestamp unchanged, causing this job to come back to this account immediately.
```suggestion
				nextCheckDuration = 1 * time.Hour // account will be deleted -> defer next check until probably after it was deleted
```